### ISSUE TITLE
    vscode-extensions.vadimcn.vscode-lldb fails to build on aarch64-darwin

### ISSUE THREAD

**Issue by Rasphino** [NixOS/nixpkgs#202507]
### Steps To Reproduce
1. run `nix build github:nixos/nixpkgs/nixpkgs-unstable#vscode-extensions.vadimcn.vscode-lldb`

### Build log
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/rfyfxfw9alm3lfbh37b94r7idia3blvy-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/hpb7avwnm52hp9klhxq75c02ihf5gkw6-cmake-build-extension-only.patch
patching file CMakeLists.txt
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/share/doc/CodeLLDB -DCMAKE_INSTALL_INFODIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/include -DCMAKE_INSTALL_SBINDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/lqiqj6s5jx196bpx0mr53d232b57y768-clang-wrapper-11.1.0/bin/strip -DCMAKE_RANLIB=/nix/store/lqiqj6s5jx196bpx0mr53d232b57y768-clang-wrapper-11.1.0/bin/ranlib -DCMAKE_AR=/nix/store/lqiqj6s5jx196bpx0mr53d232b57y768-clang-wrapper-11.1.0/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/ih46mjidpy6baf9klax4jj0drikqd17z-vscode-extension-vadimcn-vscode-lldb-1.7.4 -DVERSION_SUFFIX=
-- The C compiler identification is Clang 11.1.0
-- The CXX compiler identification is Clang 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/lqiqj6s5jx196bpx0mr53d232b57y768-clang-wrapper-11.1.0/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/lqiqj6s5jx196bpx0mr53d232b57y768-clang-wrapper-11.1.0/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Version 1.7.4
-- Could NOT find Python3 (missing: Python3_EXECUTABLE Interpreter)
-- The C compiler identification is Clang 11.1.0
-- The CXX compiler identification is Clang 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/lqiqj6s5jx196bpx0mr53d232b57y768-clang-wrapper-11.1.0/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/lqiqj6s5jx196bpx0mr53d232b57y768-clang-wrapper-11.1.0/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/nix-build-vscode-extension-vadimcn-vscode-lldb-1.7.4.drv-1/source/build/debuggee
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_NAME_DIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /tmp/nix-build-vscode-extension-vadimcn-vscode-lldb-1.7.4.drv-1/source/build
cmake: enabled parallel building
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j8 SHELL=/nix/store/49wn01k9yikhjlxc1ym5b6civ29zz3gv-bash-5.1-p16/bin/bash vsix_bootstrap
[ 12%] Generating extension.js

> vscode-lldb@1.7.4 webpack
> webpack --mode=production --stats=minimal --output-path=/tmp/nix-build-vscode-extension-vadimcn-vscode-lldb-1.7.4.drv-1/source/build --output-filename=extension.js /tmp/nix-build-vscode-extension-vadimcn-vscode-lldb-1.7.4.drv-1/source/extension/main.ts

npm ERR! code EROFS
npm ERR! syscall mkdir
npm ERR! path /homeless-shelter
npm ERR! errno -30
npm ERR! rofs EROFS: read-only file system, mkdir '/homeless-shelter'
npm ERR! rofs Often virtualized file systems, or other file systems
npm ERR! rofs that don't support symlinks, give this error.

npm ERR! Log files were not written due to an error writing to the directory: /homeless-shelter/.npm/_logs
npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
make[3]: *** [CMakeFiles/extension.dir/build.make:91: extension.js] Error 226
make[2]: *** [CMakeFiles/Makefile2:175: CMakeFiles/extension.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:263: CMakeFiles/vsix_bootstrap.dir/rule] Error 2
make: *** [Makefile:231: vsix_bootstrap] Error 2
1 asset
43 modules
webpack 5.73.0 compiled successfully in 2458 ms
```

### Notify maintainers

@Artturin @NickHu 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.1.0, macOS 13.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.0`
 - channels(rasp): `""`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by Artturin** [comment]
adding `export HOME=$TMPDIR` in preBuild should fix it

**Comment by Rasphino** [comment]
> adding `export HOME=$TMPDIR` in preBuild should fix it

Thanks! This works!

**Comment by Artturin** [comment]
Make a pr?

**Comment by superherointj** [comment]
Fixed by #211321

Re-open in case isn't fixed.
